packages/components-js/tests/e2e/puppeteer/specs/pin-code.e2e.ts

```diff
@@ -0,0 +1,645 @@
+
+    page.keyboard.press('Delete');
+    await waitForStencilLifecycle(page);
+
+    expect((await getEventSummary(host, 'update')).counter, 'after delete').toBe(1);
+    expect(await getActiveElementsAriaLabelInShadowRoot(host)).toBe('1-4');
+    expect((await getEventSummary(host, 'update')).details, 'after delete').toEqual([
+      {
+        isComplete: false,
+        value: ' 234',
+      },
+    ]);
+
+    page.keyboard.press('Delete');
+    await waitForStencilLifecycle(page);
+
+    expect((await getEventSummary(host, 'update')).counter, 'after delete').toBe(2);
+    expect(await getActiveElementsAriaLabelInShadowRoot(host)).toBe('2-4');
+    expect((await getEventSummary(host, 'update')).details, 'after delete').toEqual([
+      {
+        isComplete: false,
+        value: ' 234',
+      },
+      {
+        isComplete: false,
+        value: '  34',
+      },
+    ]);
+  });
+});
+
+describe('events', () => {
+  describe('onInput', () => {
+    it('should spread value over input elements and focus last input element', async () => {
+      await initPinCode();
+      const host = await getHost();
+      const input1 = await getInput(1);
+      const input2 = await getInput(2);
+      const input3 = await getInput(3);
+      const input4 = await getInput(4);
+      await addEventListener(input4, 'focus');
+
+      expect((await getEventSummary(input4, 'focus')).counter).toBe(0);
+
+      await input1.type('1234');
+      await waitForStencilLifecycle(page);
+
+      expect(await getProperty(input1, 'value')).toBe('1');
+      expect(await getProperty(input2, 'value')).toBe('2');
+      expect(await getProperty(input3, 'value')).toBe('3');
+      expect(await getProperty(input4, 'value')).toBe('4');
+      expect(await getActiveElementsAriaLabelInShadowRoot(host)).toBe('4-4');
+      expect(await getProperty(host, 'value')).toStrictEqual('1234');
+    });
+
+    it('should spread value over input elements and focus last empty input element if value is too short', async () => {
+      await initPinCode();
+      const host = await getHost();
+      const input1 = await getInput(1);
+      const input2 = await getInput(2);
+      const input3 = await getInput(3);
+      const input4 = await getInput(4);
+      await addEventListener(input4, 'focus');
+
+      expect((await getEventSummary(input4, 'focus')).counter).toBe(0);
+
+      await input1.type('12');
+      await waitForStencilLifecycle(page);
+
+      expect(await getProperty(input1, 'value')).toBe('1');
+      expect(await getProperty(input2, 'value')).toBe('2');
+      expect(await getProperty(input3, 'value')).toBe('');
+      expect(await getProperty(input4, 'value')).toBe('');
+      expect(await getActiveElementsAriaLabelInShadowRoot(host)).toBe('3-4');
+      expect(await getProperty(host, 'value')).toStrictEqual('12  ');
+    });
+  });
+
+  // TODO: onPaste event -> puppeteer support for onPaste event is currently very low
```

Comment:
So dispatching the event manually didn't work?

Reply:
Why not? I think it should work like demonstrated here: https://stackblitz.com/edit/js-d6wq6b?file=index.html,index.js